test/controllers/LocationTypeControllerSpec.scala

```diff
@@ -0,0 +1,176 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.{AppWithDefaultMockFixtures, SpecBase}
+import forms.EnumerableFormProvider
+import generators.Generators
+import models.{LocationType, NormalMode}
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{reset, when}
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalacheck.Gen
+import pages.LocationTypePage
+import play.api.inject.bind
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import services.LocationTypeService
+import views.html.LocationTypeView
+
+import scala.concurrent.Future
+
+class LocationTypeControllerSpec extends SpecBase with AppWithDefaultMockFixtures with Generators {
+
+  private val lts = Gen.nonEmptyListOf(arbitrary[LocationType]).sample.value
+  private val lt  = lts.head
+
+  private val formProvider                                 = new EnumerableFormProvider()
+  private val form                                         = formProvider("locationOfGoods.locationType", lts)
+  private val mode                                         = NormalMode
+  private lazy val locationTypeRoute                       = controllers.locationOfGoods.routes.LocationTypeController.onPageLoad(departureId, mode).url
+  private val mockLocationTypeService: LocationTypeService = mock[LocationTypeService]
+
+  override protected def guiceApplicationBuilder(): GuiceApplicationBuilder =
+    super
+      .guiceApplicationBuilder()
+      .overrides(bind(classOf[LocationTypeService]).toInstance(mockLocationTypeService))
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    reset(mockLocationTypeService)
+    when(mockLocationTypeService.getLocationTypes(any())(any())).thenReturn(Future.successful(lts))
+  }
+
+  "LocationType Controller" - {
+
+    "must return OK and the correct view for a GET" in {
+
+      setExistingUserAnswers(emptyUserAnswers)
+
+      val request = FakeRequest(GET, locationTypeRoute)
+
+      val result = route(app, request).value
+
+      val view = injector.instanceOf[LocationTypeView]
+
+      status(result) mustEqual OK
+
+      contentAsString(result) mustEqual
+        view(form, departureId, lrn.toString, lts, mode)(request, messages).toString
+    }
+
+//    "must redirect to the next page and infer LocationType when only one location type" in {
+//
+//      when(mockLocationTypeService.getLocationTypes(any())(any())).thenReturn(Future.successful(Seq(lt)))
+//
+//      val userAnswers = emptyUserAnswers.setValue(em, Simplified)
+//      setExistingUserAnswers(emptyUserAnswers)
```

Comment:
Can we remove this test if we don't need it?

Reply:
Hey Jake, I have commented this out as its dependant on another ticket in sprint [CTCP-3844](https://jira.tools.tax.service.gov.uk/browse/CTCP-3844). Once that is implemented I can introduce this test and the logic back in. Happy to explain over a call if need be. I wanted to get this merged at this stage as there is some stuff in the navigator that can help some other tickets :)